tests/test_collections.py

```diff
@@ -157,5 +227,57 @@ def test_collections(self):
                                params=dict(replica="aws"))
             self.assertEqual(res.status_code, requests.codes.not_found)
 
+    def upload_file(self, contents):
```

Comment:
Consider maintaining authorship and keeping the diff small by moving this back to the original place.